File: keygen.js
function generateKey() {
  const password = "YourPass";
  const xorValue = 10;
  let xorOp = "";

  for (let i = 0; i < password.length; i++) {
    xorOp += String.fromCharCode(password.charCodeAt(i) ^ xorValue);
  }

  let keygen = "";
  for (let i = 0; i < xorOp.length; i++) {
    keygen += String.fromCharCode(xorOp.charCodeAt(i) ^ xorValue);
  }

  console.log("Mot de passe obfusqué :", xorOp);
  console.log("Clé générée (keygen) :", keygen);
}

generateKey();

File: keygen.c
#include <stdio.h>
#include <string.h>

int main() {
    char xorOp[9];
    char password[9];
    char keygen[9];

    for(int i = 0 ; i < 8; i + 1){
        xorOp[i] = password[i] ^ 10;
    }
    xorOp[8] = '\0';

    for(int i = 0; i < 8; i++){
        keygen[i] = xorOp[i] ^ 10;
    }
    keygen[8] = '\0';

    printf("Clé générée : %s\n", keygen);

    return 0;
}